### ISSUE TITLE
allshortestPaths error (start/end nodes the same) with cypher.forbid_shortestpath_common_node=false

### ISSUE THREAD

**Issue by Alan-R** [neo4j/neo4j#9992]
I'm getting an error message when running the query below:

```
MATCH p = allshortestPaths( (start:Class_Drone)-[:tcpclient|:tcpservice|:hosting|:baseip|:nicowner|:monitoring|:wiredto|:ipowner*]-(m) )
        WHERE m.nodetype IN ['IPaddrNode', 'SystemNode', 'MonitorAction', 'ProcessNode', 'IPtcpportNode', 'Drone', 'NICNode']
        UNWIND nodes(p) AS n
        UNWIND rels(p) AS r
        RETURN [x in COLLECT(DISTINCT n) WHERE x.nodetype in ['IPaddrNode', 'SystemNode', 'MonitorAction', 'ProcessNode', 'IPtcpportNode', 'Drone', 'NICNode']] AS nodes,
        COLLECT(DISTINCT r) AS relationships
```
I have this line in my neo4j config file:
**cypher.forbid_shortestpath_common_node=false**

And I still get this message:
The shortest path algorithm does not work when the start and end nodes are the same. This can happen if you perform a shortestPath search after a cartesian product that might have the same start and end nodes for some of the rows passed to shortestPath. If you would rather not experience this exception, and can accept the possibility of missing results for those rows, disable this in the Neo4j configuration by setting `cypher.forbid_shortestpath_common_node` to false. If you cannot accept missing results, and really want the shortestPath between two common nodes, then re-write the query using a standard Cypher variable length pattern expression followed by ordering by path length and limiting to one result.

Syslog adds this phrase:
reference 78c83d45-4e40-4774-a6be-e347907e7ec9

FWIW, ***this query used to work in earlier (3.0.x) versions of Neo4j***. The intent of the query is to return the selected portions of the graph  as a subgraph. This particular one returns basically the entire graph, but other variations are more selective. They all get the same message...

Suggestions on how to work around this would be appreciated. I have to say that after reading the message, I'm not certain I understand it. It could be clearer. It's also pretty clear that being able to select and return a subgraph is a pretty important capability. If this isn't the right way, then tell me the right way...

### Operating system:
    "operating-system": "GNU/Linux",
    "machine": "x86_64",
    "processor": "x86_64",
    "hardware-platform": "x86_64",
    "kernel-name": "Linux",
    "kernel-release": "4.4.0-92-generic",
    "kernel-version": "#115-Ubuntu SMP Thu Aug 10 09:04:33 UTC 2017",
    "Distributor ID":   "Ubuntu",
    "Description":      "Ubuntu 16.04.3 LTS",
    "Release":  "16.04",
    "Codename": "xenial"

### API/Driver:
Both using py2neo and using the console
### **Steps to reproduce**
Run queries like the one I shared above
### Expected behavior:
Get one row with two columns - one of nodes, and one of relationships
### Actual behavior
Got the error described above

**Comment by Alan-R** [comment]
I did restart the server after adding the aforementioned setting. I did it several times, and I verified that the running Neo4j was looking at that particular config file (by looking at ps -ef).

**Comment by jexp** [comment]
Did you try to set the setting to true instead? B/c it is a "common-node"

**Comment by mneedham** [comment]
I think you're missing an 's' on the config name. Can you try this instead:

> `cypher.forbid_shortestpath_common_nodes=false`

**Comment by Alan-R** [comment]
I certainly can. I did cut and paste the name from the message into the
settings file.
--
  Alan Robertson
  alanr@unix.sh



On Mon, Sep 11, 2017, at 04:19 AM, Michael Hunger wrote:
> Did you try to set the setting to true instead? B/c it is a "common-
> node"> — You are receiving this because you authored the thread. Reply to
> this email directly, view it on GitHub[1], or mute the thread[2].> 


Links:

  1. https://github.com/neo4j/neo4j/issues/9992#issuecomment-328487120

**Comment by Alan-R** [comment]
I tried it, and it didn't complain. Sorry for my mistake causing you trouble.
I can't yet tell if the query still does what it's supposed to do, but this eliminates the cause for this bug report - so I'll close it.

Many thanks for your help!

**Comment by szarnyasg** [comment]
Version 3.3.1 still displays the error message as `_node` and not `_nodes`:

```
Neo.DatabaseError.Statement.ExecutionFailed
The shortest path algorithm does not work when the start and end nodes are the same. This can happen if you
perform a shortestPath search after a cartesian product that might have the same start and end nodes for some
of the rows passed to shortestPath. If you would rather not experience this exception, and can accept the
possibility of missing results for those rows, disable this in the Neo4j configuration by setting
`cypher.forbid_shortestpath_common_node` to false. If you cannot accept missing results, and really want the
shortestPath between two common nodes, then re-write the query using a standard Cypher variable length pattern
expression followed by ordering by path length and limiting to one result.
```

**Comment by Alan-R** [comment]
Please accept this pull request which fixes the problem: https://github.com/neo4j/neo4j/pull/10988

**Comment by eebus** [comment]
Seems to have been fixed with #10522

**Comment by Th3-Muffin-Man** [comment]
Alternatively you can include the following instruction (WHERE n<>r) to ensure that the End Nodes and Start Nodes are Not the same:

```
MATCH p = allshortestPaths( (start:Class_Drone)-[:tcpclient|:tcpservice|:hosting|:baseip|:nicowner|:monitoring|:wiredto|:ipowner*]-(m) )
        WHERE m.nodetype IN ['IPaddrNode', 'SystemNode', 'MonitorAction', 'ProcessNode', 'IPtcpportNode', 'Drone', 'NICNode']
        UNWIND nodes(p) AS n
        UNWIND rels(p) AS r
        WHERE n<>r
        RETURN [x in COLLECT(DISTINCT n) WHERE x.nodetype in ['IPaddrNode', 'SystemNode', 'MonitorAction', 'ProcessNode', 'IPtcpportNode', 'Drone', 'NICNode']] AS nodes,
        COLLECT(DISTINCT r) AS relationships
```
